WordPress/Classes/ViewRelated/System/Notices/NoticePresenter.swift

```diff
@@ -79,23 +114,49 @@ class NoticePresenter: NSObject {
         }
     }
 
-    private func presentNextNoticeIfAvailable() {
-        if let notice = store.nextNotice {
-            present(notice)
+    /// Handle all changes in the `NoticeStore`.
+    ///
+    /// In here, we determine whether to show a Notice or dismiss the currently shown Notice based
+    /// on the value of `NoticeStore.currentNotice`.
+    private func onStoreChange() {
+        guard currentNoticePresentation?.notice != store.currentNotice else {
+            return
+        }
+
+        dismissForegroundNotice()
+
+        currentNoticePresentation = nil
+
+        guard let notice = store.currentNotice else {
+            return
+        }
+
+        if let artifact = present(notice) {
```

Comment:
I'm not sure if you finished the artifact renaming changes yet, but we may as well rename this to `presentation` too, for consistency :)

Reply:
Good find 😁